### ISSUE TITLE
Error In Console "Could not pass event AsyncPlayerChatEvent"

### ISSUE THREAD

**Issue by PAINmatrix** [tastybento/ASkyBlock-Bugs-N-Features#241]
Any time a player talks in the defined ASkyBlock world it sends this error:
`[12:04:40] [Async Chat Thread - #1/ERROR]: **** THIS IS NOT AN ISSUE TO DO WITH PER WORLD PLUGINS ****
Could not pass event AsyncPlayerChatEvent to ASkyBlock v3.0.6.6
org.bukkit.event.EventException
	at us.Myles.PWP.TransparentListeners.PerWorldPluginLoader$1.execute(PerWorldPluginLoader.java:122) ~[?:?]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at us.Myles.PWP.TransparentListeners.PWPRegisteredListener.callEvent(PWPRegisteredListener.java:30) [PerWorldPlugins-1.11.jar:?]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at net.minecraft.server.v1_11_R1.PlayerConnection.chat(PlayerConnection.java:1266) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at net.minecraft.server.v1_11_R1.PlayerConnection.a(PlayerConnection.java:1204) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at net.minecraft.server.v1_11_R1.PacketPlayInChat$1.run(PacketPlayInChat.java:39) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.util.MissingFormatArgumentException: Format specifier '%32132$32s'
	at org.bukkit.event.player.AsyncPlayerChatEvent.setFormat(AsyncPlayerChatEvent.java:100) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
	at com.wasteofplastic.askyblock.listeners.ChatListener.onChat(ChatListener.java:111) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at us.Myles.PWP.TransparentListeners.PerWorldPluginLoader$1.execute(PerWorldPluginLoader.java:120) ~[?:?]
	... 12 more`
Although this error does not seem to affect anything it is  a nuisance...

**Comment by Poslovitch** [comment]
This error isn't thrown by ASkyBlock but by «PWP» which says : `THIS IS NOT AN ISSUE TO DO WITH PER WORLD PLUGINS ***`
Try removing this plugin.

**Comment by PAINmatrix** [comment]
`23.05 19:51:37 [Server] ERROR Could not pass event AsyncPlayerChatEvent to ASkyBlock v3.0.6.6
23.05 19:51:37 [Server] INFO org.bukkit.event.EventException
23.05 19:51:37 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at net.minecraft.server.v1_11_R1.PlayerConnection.chat(PlayerConnection.java:1266) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at net.minecraft.server.v1_11_R1.PlayerConnection.a(PlayerConnection.java:1204) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at net.minecraft.server.v1_11_R1.PacketPlayInChat$1.run(PacketPlayInChat.java:39) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
23.05 19:51:37 [Server] INFO at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
23.05 19:51:37 [Server] INFO at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
23.05 19:51:37 [Server] INFO at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
23.05 19:51:37 [Server] INFO at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
23.05 19:51:37 [Server] INFO Caused by: java.util.FormatFlagsConversionMismatchException: Conversion = s, Flags = 0
23.05 19:51:37 [Server] INFO at org.bukkit.event.player.AsyncPlayerChatEvent.setFormat(AsyncPlayerChatEvent.java:100) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO at com.wasteofplastic.askyblock.listeners.ChatListener.onChat(ChatListener.java:111) ~[?:?]
23.05 19:51:37 [Server] INFO at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
23.05 19:51:37 [Server] INFO at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
23.05 19:51:37 [Server] INFO at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
23.05 19:51:37 [Server] INFO at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
23.05 19:51:37 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:37 [Server] INFO ... 11 more
23.05 19:51:37 [Server] INFO ❤ <Hub>  [CoOwner] Inferno: we're going to have to restart again in a minute or so. i need to grab an error.
23.05 19:51:40 [Server] INFO _Inferno_X issued server command: /skyblock
23.05 19:51:40 [Server] INFO _Inferno_X [b126f823-988f-477e-9d85-1a8d383f1f78]: /skyblock
23.05 19:51:40 [Server] INFO Title command successfully executed
23.05 19:51:43 [Server] ERROR Could not pass event AsyncPlayerChatEvent to ASkyBlock v3.0.6.6
23.05 19:51:43 [Server] INFO org.bukkit.event.EventException
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at net.minecraft.server.v1_11_R1.PlayerConnection.chat(PlayerConnection.java:1266) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at net.minecraft.server.v1_11_R1.PlayerConnection.a(PlayerConnection.java:1204) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at net.minecraft.server.v1_11_R1.PacketPlayInChat$1.run(PacketPlayInChat.java:39) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO Caused by: java.util.FormatFlagsConversionMismatchException: Conversion = s, Flags = 0
23.05 19:51:43 [Server] INFO at org.bukkit.event.player.AsyncPlayerChatEvent.setFormat(AsyncPlayerChatEvent.java:100) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at com.wasteofplastic.askyblock.listeners.ChatListener.onChat(ChatListener.java:111) ~[?:?]
23.05 19:51:43 [Server] INFO at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO ... 11 more`

Removed, and the same error.

**Comment by PAINmatrix** [comment]
`23.05 19:51:43 [Server] ERROR Could not pass event AsyncPlayerChatEvent to ASkyBlock v3.0.6.6
23.05 19:51:43 [Server] INFO org.bukkit.event.EventException
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at net.minecraft.server.v1_11_R1.PlayerConnection.chat(PlayerConnection.java:1266) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at net.minecraft.server.v1_11_R1.PlayerConnection.a(PlayerConnection.java:1204) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at net.minecraft.server.v1_11_R1.PacketPlayInChat$1.run(PacketPlayInChat.java:39) [spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
23.05 19:51:43 [Server] INFO Caused by: java.util.FormatFlagsConversionMismatchException: Conversion = s, Flags = 0
23.05 19:51:43 [Server] INFO at org.bukkit.event.player.AsyncPlayerChatEvent.setFormat(AsyncPlayerChatEvent.java:100) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO at com.wasteofplastic.askyblock.listeners.ChatListener.onChat(ChatListener.java:111) ~[?:?]
23.05 19:51:43 [Server] INFO at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
23.05 19:51:43 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot83.jar:git-Spigot-b32c8f8-4d3bf20]
23.05 19:51:43 [Server] INFO ... 11 more`

Removed, and same error.

**Comment by Poslovitch** [comment]
Okay, so PWP was just handling the error.
Let me have a look

**Comment by tastybento** [comment]
@PAINmatrix It's something to do with what the chat prefix settings in config.yml. What do you have set for these?

This is the default:

```
   # Variables to use as chat prefixes. You can change them to be what you want.
   # {ISLAND_LEVEL} is replaced by the player's island level.
   # {ISLAND_CHALLENGE_LEVEL} is replaced by the player's challenge level.
   # {ISLAND_PLAYER} is replaced by the player's display name
   # Confirmed to work with Essentials Chat, Multiverse Core and Herochat.
   # Do not use these in PEX prefixes - it doesn't work.
   chatlevelprefix: "[ISLAND_LEVEL]"
   chatchallengelevelprefix: "{ISLAND_CHALLENGE_LEVEL}"
   chatislandplayer: "{ISLAND_PLAYER}"
   # Displays the island level {ISLAND_LEVEL} with a XX.Xk pattern in chat
   # e.g : [23.5k] instead of [23562]
   fancylevelinchat: false
`

**Comment by PAINmatrix** [comment]
`   # Variables to use as chat prefixes. You can change them to be what you want.
   # {ISLAND_LEVEL} is replaced by the player's island level.
   # {ISLAND_CHALLENGE_LEVEL} is replaced by the player's challenge level.
   # {ISLAND_PLAYER} is replaced by the player's display name
   # Confirmed to work with Essentials Chat, Multiverse Core and Herochat.
   # Do not use these in PEX prefixes - it doesn't work.
   chatlevelprefix: ""
   chatchallengelevelprefix: ""
   chatislandplayer: ""
   # Displays the island level {ISLAND_LEVEL} with a XX.Xk pattern in chat
   # e.g : [23.5k] instead of [23562]
   fancylevelinchat: false`

**Comment by PAINmatrix** [comment]
`
   # Variables to use as chat prefixes. You can change them to be what you want.
   # {ISLAND_LEVEL} is replaced by the player's island level.
   # {ISLAND_CHALLENGE_LEVEL} is replaced by the player's challenge level.
   # {ISLAND_PLAYER} is replaced by the player's display name
   # Confirmed to work with Essentials Chat, Multiverse Core and Herochat.
   # Do not use these in PEX prefixes - it doesn't work.
   chatlevelprefix: ""
   chatchallengelevelprefix: ""
   chatislandplayer: ""
   # Displays the island level {ISLAND_LEVEL} with a XX.Xk pattern in chat
   # e.g : [23.5k] instead of [23562]
   fancylevelinchat: false
`

**Comment by PAINmatrix** [comment]
```
   # Variables to use as chat prefixes. You can change them to be what you want.
   # {ISLAND_LEVEL} is replaced by the player's island level.
   # {ISLAND_CHALLENGE_LEVEL} is replaced by the player's challenge level.
   # {ISLAND_PLAYER} is replaced by the player's display name
   # Confirmed to work with Essentials Chat, Multiverse Core and Herochat.
   # Do not use these in PEX prefixes - it doesn't work.
   chatlevelprefix: ""
   chatchallengelevelprefix: ""
   chatislandplayer: ""
   # Displays the island level {ISLAND_LEVEL} with a XX.Xk pattern in chat
   # e.g : [23.5k] instead of [23562]
   fancylevelinchat: false
```

**Comment by tastybento** [comment]
I see the problem - the empty string is causing the error. Quick fix, put those strings back to the defaults. Longer fix is in the code for 3.0.6.8. Thanks for the report.
